sympy/solvers/polysys.py

```diff
@@ -258,7 +267,16 @@ def _solve_reduced_system(system, gens, entry=False):
         gens = f.gens
         gen = gens[-1]
 
-        zeros = list(roots(f.ltrim(gen)).keys())
+        rts = roots(f.ltrim(gen))
+
+        if sum(rts.values()) < degree(f, gen):
+            raise NotImplementedError(filldedent('''
```

Comment:
I don't think this should raise unconditionally. Sometimes it's useful to get partial solutions. Instead maybe there should be an flag to `solve_poly_system` to decide whether to raise or to return a partial solution (preserving the current behaviour).